internal/queries/queries_consumer.go

```diff
@@ -11,25 +12,97 @@ const (
 	PayloadJoins  = "left join Payloads on Payloads.id = PayloadStatuses.payload_id"
 )
 
+type PayloadFieldsRepository interface {
+	GetStatus(string) models.Statuses
+	GetService(string) models.Services
+	GetSource(string) models.Sources
+}
 
-func GetServiceByName(db *gorm.DB, service_id string) models.Services {
-	var service models.Services
-	db.Where("name = ?", service_id).First(&service)
-	return service
+type PayloadFieldsRepositoryFromDB struct {
+	DB *gorm.DB
+}
+
+type PayloadFieldsRepositoryFromCache struct {
+	DB           PayloadFieldsRepository
+	StatusCache  map[string]models.Statuses
+	ServiceCache map[string]models.Services
+	SourceCache  map[string]models.Sources
 }
 
-func GetStatusByName(db *gorm.DB, status_id string) models.Statuses {
+func (d *PayloadFieldsRepositoryFromDB) GetStatus(statusName string) models.Statuses {
 	var status models.Statuses
-	db.Where("name = ?", status_id).First(&status)
+
+	d.DB.Where("name = ?", statusName).First(&status)
 	return status
 }
 
-func GetSourceByName(db *gorm.DB, source_id string) models.Sources {
+func (d *PayloadFieldsRepositoryFromDB) GetService(serviceName string) models.Services {
+	var service models.Services
+
+	d.DB.Where("name = ?", serviceName).First(&service)
+	return service
+}
+
+func (d *PayloadFieldsRepositoryFromDB) GetSource(sourceName string) models.Sources {
 	var source models.Sources
-	db.Where("name = ?", source_id).First(&source)
+
+	d.DB.Where("name = ?", sourceName).First(&source)
 	return source
 }
 
+func (d *PayloadFieldsRepositoryFromCache) GetStatus(statusName string) models.Statuses {
+	cached, ok := d.StatusCache[statusName]
+	if ok {
+		return cached
+	}
+
+	dbEntry := d.DB.GetStatus(statusName)
+
+	d.StatusCache[statusName] = dbEntry
+
+	return dbEntry
+}
+
+func (d *PayloadFieldsRepositoryFromCache) GetService(serviceName string) models.Services {
+	cached, ok := d.ServiceCache[serviceName]
+	if ok {
+		return cached
+	}
+
+	dbEntry := d.DB.GetService(serviceName)
+
+	d.ServiceCache[serviceName] = dbEntry
+
+	return dbEntry
+}
+
+func (d *PayloadFieldsRepositoryFromCache) GetSource(sourceName string) models.Sources {
+	cached, ok := d.SourceCache[sourceName]
+	if ok {
+		return cached
+	}
+
+	dbEntry := d.DB.GetSource(sourceName)
+
+	d.SourceCache[sourceName] = dbEntry
+
+	return dbEntry
+}
+
+func NewPayloadFieldsRepository(db *gorm.DB) PayloadFieldsRepository {
```

Comment:
Let's explicitly pass in the config object to this method.

Reply:
I originally had done that, but it seemed ugly to force every caller to supply the config to it, when we really just needed the db.